libs/compress/build.gradle

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+apply plugin: 'opensearch.build'
+apply plugin: 'opensearch.publish'
+
+base {
```

Comment:
we may not run the compressor tests with security manager, needs double checking

Reply:
I think this was major concern on initial pull request (and was the concern for Apache Lucene for not getting ZSTD in). AFAIK the conclusion was that value of the feature overweight the JNI risks, and I think bundling into core would allow to use the ZSTD in snapshots (which I think is not possible with plugins at the moment).

Reply:
> I think bundling into core allowed to use the ZSTD in snapshots (more widely, repositories, which I think is not possible with plugins at the moment).

With this SPI implementation we can install it as a plugin an it's available to all plugins. This also solves the portability risk. If someone ends up installing OpenSearch on an incompatible platform / OS / glibc they're hosed today. With zstd as an optional plugin they're not, they just don't use the zstd plugin and stick wtith Deflate.

Reply:
> With this SPI implementation we can install it as a plugin an it's available to all plugins

Correct

> If someone ends up installing OpenSearch on an incompatible platform / OS / glibc they're hosed today. 

Also agree

Reply:
> We didn't introduce zstd as experimental either, 

We did initially, it was contributed as a sandbox plugin (== experimental), but the experimental label was dropped once it has been moved to core.

Reply:
> ...but the experimental label was dropped once it has been moved to core.

Yeah, this is what I meant. Probably should've kept the `@experimental` label even after moving from sandbox to `:server`.